src/debug.ts

```diff
@@ -114,3 +114,101 @@ export class PreLaunchTaskProvider implements vscode.TaskProvider
+    }
+
+    async resolveDebugConfiguration(folder : vscode.WorkspaceFolder | undefined, config : vscode.DebugConfiguration, token : vscode.CancellationToken | undefined) : Promise<vscode.DebugConfiguration>
+    {
+        return config;
+    }
+
+    protected async resolvePackage(folder: vscode.WorkspaceFolder | undefined) : Promise<packages.RockPackage | undefined>
+    {
+        if (!folder) {
+            return;
+        }
+        let pkg = await this._context.getPackageByPath(folder.uri.fsPath);
+        if (pkg instanceof packages.RockPackage) {
+            return pkg;
+        }
+    }
+}
+
+export class CXXConfigurationProvider extends ConfigurationProvider
+{
+    provideDebugConfigurations(folder : vscode.WorkspaceFolder | undefined, token : vscode.CancellationToken | undefined) : vscode.ProviderResult<vscode.DebugConfiguration[]> {
+        return [];
+    }
+
+    async resolveDebugConfiguration(folder : vscode.WorkspaceFolder | undefined, config : vscode.DebugConfiguration, token : vscode.CancellationToken | undefined) : Promise<vscode.DebugConfiguration>
+    {
+        let pkg = await this.resolvePackage(folder);
+        if (!pkg) {
+            return config;
+        }
+
+        let ws = pkg.ws;
+
+        let debuggerPath = config.miDebuggerPath || config.MIMode;
+        let stubScript = joinpath(__dirname, '..', 'stubs', config.MIMode);
+
+        config.miDebuggerPath = stubScript;
+        if (!config.environment) {
+            config.environment = [];
+        }
+        config.environment = config.environment.concat([
+            { name: "VSCODE_ROCK_AUTOPROJ_PATH", value: ws.autoprojExePath() },
+            { name: "VSCODE_ROCK_AUTOPROJ_DEBUGGER", value: debuggerPath },
+            { name: 'AUTOPROJ_CURRENT_ROOT', value: ws.root }
+        ])
+
+        if (!fs.existsSync(config.program)) {
+            let search = [pkg.info.builddir, pkg.info.prefix,
+                joinpath(pkg.info.builddir, 'test'),
+                joinpath(pkg.info.builddir, 'src'),
+                joinpath(pkg.info.prefix, 'bin')]
+            let foundPath = search.find((dir) => {
+                return fs.existsSync(joinpath(dir, config.program));
+            })
+            if (foundPath) {
+                config.program = joinpath(foundPath, config.program);
+            }
+        }
+        if (!config.cwd) {
+            config.cwd = dirname(config.program);
+        }
+        return config;
+    }
+}
+
+export class RubyConfigurationProvider extends ConfigurationProvider
+{
+    async resolveDebugConfiguration(folder : vscode.WorkspaceFolder | undefined, config : vscode.DebugConfiguration, token : vscode.CancellationToken | undefined) : Promise<vscode.DebugConfiguration>
+    {
+        let pkg = await this.resolvePackage(folder);
+        if (!pkg) {
+            return config;
+        }
+        let ws = pkg.ws;
+
+        config.useBundler = true;
+        config.pathToBundler = ws.autoprojExePath();
```

Comment:
Just saw this when I was playing with this stuff yesterday. Found two issues:

1. One needs rdebug-ide installed in the workspace;
2. There is a bug in the ruby extension (we need an extra '--' [here](https://github.com/rubyide/vscode-ruby/blob/master/src/debugger/ruby.ts#L203) otherwise `rdebug-ide` will swallow arguments

But, still better than the EnvironmentBridge thing...

Reply:
There is also `pathToRDebug`.. We could use that instead and have a stub/wrapper for rdebug-ide as well

Reply:
> One needs rdebug-ide installed in the workspace;

Well, we'll need to have a bunch of things installed in the workspace. My plan there is to create a [rock.vscode](https://github.com/rock-core/rock.vscode-package_set) package set that people have to add to their workspace if they want to use the extension.

> There is a bug in the ruby extension (we need an extra '--' here otherwise rdebug-ide will swallow arguments

I haven't seen that problem (i.e. worked for me).

> Just saw this when I was playing with this stuff yesterday.

I kind of feel that we're basically spending both our times on the same things. If you want to do it, I'd be happy to let you, but I really feel wasting my time if you're doing it at the same time anyways.

Reply:
> My plan there is to create a rock.vscode package set that people have to add to their workspace if they want to use the extension.

👍 

> I haven't seen that problem (i.e. worked for me).

Try debuging `syskit ide -rdefault`

> If you want to do it, I'd be happy to let you

I do not.. I just wanted to see how the workflow would feel...

Reply:
>  but I really feel wasting my time if you're doing it at the same time anyways.

If you want to help to get this done quicker, there are a few things that would:
- generation of the launch.json file entries
- running a Syskit instance (on a non-standard port to avoid conflict with instances the user might have started himself) behind the scenes and discover orogen models this way
- discover CMake binaries (in its simplest form, just listing executables from `${prefix}/bin` as listed in the CMake installation manifest, and `${builddir}/test` would be enough IMO).

Reply:
I'm on the first one already.. Are we keeping the package selection mechanism? If we are not, we will need an extra picker to let the user choose to what package he/she wants the entries to be added

Reply:
> Are we keeping the package selection mechanism? 

Not sure. What do you think ?

Reply:
Not sure either.. If we are both unsure I would say we leave it there until we are sure it is not really useful anymore

Reply:
That's my policy for now, I keep the ForeignPackage / selection mechanism / debug mechanism until the new stuff is functional and proven to be useful

It's going to be weird for a while, but that's what 0.x versions are for.